webcompat/api/uploads.py

```diff
@@ -58,10 +58,12 @@ def to_image_object(self, imagedata):
             if isinstance(imagedata, FileStorage):
                 return Image.open(imagedata)
             # Is this a base64 encoded image (i.e., bug report screenshot)?
-            if (isinstance(imagedata, unicode) and
+            if (isinstance(imagedata, str) and
                     imagedata.startswith('data:image/')):
                 # Chop off 'data:image/.+;base64,' before decoding
                 imagedata = re.sub('^data:image/.+;base64,', '', imagedata)
+                # This will fix any incorrectly padded data.
```

Comment:
@miketaylr 

> In theory, the padding character is not needed for decoding, since the number of missing bytes can be calculated from the number of Base64 digits. In some implementations, the padding character is mandatory, while for others it is not used. One case in which padding characters are required is concatenating multiple Base64 encoded files. 
— https://en.wikipedia.org/wiki/Base64#Output_padding

See also https://gist.github.com/perrygeo/ee7c65bb1541ff6ac770


> In the above scenario, the input character string contains 48 bits, an exact multiple of 24 that enables exact grouping into 6 bit groups. But in the event that there are less than 24 characters, or if the number of bits in the input stream is not a multiple of 24, padding is used to make up for the remaining bits.
> 
> The methodology for padding is as follows.
> 
> Add as many “zero” bits to the right of the 6-bit grouped bit stream so that the total bit length is a multiple of 24. If the modified input data contains any octets that contain only padded zeroes, replace each of those octets with the padding character “=”. 

https://blogs.oracle.com/rammenon/base64-explained

```python
imagedata = imagedata + '==='
```

So basically by adding 3 characters we will address all scenarios, that requires a multiple of 4. The padding character is `=`, and it will be ignored for things longer than multiple of 4.

Let's say we have: `1234567` aka 7 characters. We miss one character to be able to decode that.

```
>>> import base64
>>> base64.b64decode('1234567')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/karl/.virtualenvs/koke/lib/python3.7/base64.py", line 87, in b64decode
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding
>>> base64.b64decode('1234567=')
b'\xd7m\xf8\xe7\xae'
>>> base64.b64decode('1234567==')
b'\xd7m\xf8\xe7\xae'
>>> base64.b64decode('1234567===')
b'\xd7m\xf8\xe7\xae'
>>> base64.b64decode('1234567====')
b'\xd7m\xf8\xe7\xae'
>>> base64.b64decode('1234567=====')
b'\xd7m\xf8\xe7\xae'
```